ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/UserManagementServiceIT.scala

```diff
@@ -178,6 +180,68 @@ final class UserManagementServiceIT extends LedgerTestSuite {
     } yield ()
   })
 
+  userManagementTest(
+    "CreateUsersRaceCondition",
+    "Tests scenario of multiple concurrent create-user calls for the same user",
+    runConcurrently = false,
+  ) {
+    implicit ec =>
+      { participant =>
+        val attempts = (1 to 10).toVector
+        val userId = participant.nextUserId()
+        val request =
+          CreateUserRequest(Some(User(id = userId, primaryParty = "")), rights = Seq.empty)
+        for {
+          results <- Future
+            .traverse(attempts) { _ =>
+              participant.createUser(request).transform(Success(_))
+            }
+        } yield {
+          assertSingleton(
+            "successful user creation",
+            results.filter(_.isSuccess),
+          )
+          val unexpectedErrors = results
+            .collect { case x if x.isFailure => x.failed.get }
+            .filterNot(t =>
+              LedgerApiErrors.AdminServices.UserAlreadyExists.matches(t) || ErrorDetails
+                .isInternalError(t)
```

Comment:
When do we get an internal error?

Reply:
On db unique constraint violation

Reply:
Can we make this error deterministic?

Reply:
I am not aware of the low-level details of the DB queries for this use-case, but wouldn't something like `INSERT.. ON CONFLICT DO NOTHING` and checking the returned value give us the basis for constructing the expected error on duplicate?

Reply:
`INSERT.. ON CONFLICT DO NOTHING` worked with Postgres but it wasn't obvious in Oracle
@nmarton-da have we figured out to replicate `INSERT.. ON CONFLICT DO NOTHING` in Oracle?
No idea about H2.

It's enticing to not assert receiving an internal error but we would likely have to pay a cost of diverging backend implementations.

Reply:
Discussed offline. I don't think it's sane to return `INTERNAL` to the user easily on admin requests, but I'm fine to let it go if it's for a performance benefit/code simplicity. CC @nmarton-da @mziolekda

Reply:
**TLDR**: with Pawel we discussed this back in the day, and this tradeoff was struck to 
* favor the majority of the use cases, 
* over a fragile/complex heuristic favoring a minor usage as well.

The fundamental idea behind this approach was discussed at the inception of the UserManagementStore:
* Optimistic query and set mechanics providing nice error messages without DB race (majority of cases)
* Providing general retry-able error (I guess that is the internal error case here) for DB races (expected to be vast minority of the cases)
* Please note there is no compromise on correctness here. It is only about nice error messages.

This tradeoff allowed not caring about special mechanics how we can achieve nice error messages in DB race cases.
Additional note: ON CONFLICT DO NOTHING works only nice for PostgreSQL, on Oracle it is a buggy bleeding edge, and on H2 for sure it will boil down to exception parsing to our best knowledge.

Reply:
Alright, since you're saying this cannot be easily triggered in prod use-cases, I'm fine with it like this.

Reply:
@nmarton-da 

> Providing general retry-able error (I guess that is the internal error case here) for DB races (expected to be vast minority of the cases)

It's not retry-able. It is an opaque internal error without any details. The error code (visible only on the server side) is `INDEX_DB_SQL_NON_TRANSIENT_ERROR`


Fun fuct, on H2 contraint violation actually gets mapped to a retryable `INDEX_DB_SQL_TRANSIENT_ERROR`

Reply:
Yep, on H2 there's a caveat: https://github.com/digital-asset/daml/blob/main/ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/DatabaseSelfServiceError.scala#L32-L36